testing/client.related.test.js

```diff
@@ -3,114 +3,257 @@ import '@testing-library/jest-dom/extend-expect';
 import nock from 'nock';
 import React from 'react';
 import { BrowserRouter as Router } from 'react-router-dom';
-// import App from '../client/src/components/App.jsx';
-// import Related from '../client/src/components/Related/Related.jsx';
+import App from './mockApp.jsx';
 import RelatedProducts from '../client/src/components/Related/RelatedProducts.jsx';
-// import YourOutfit from '../client/src/components/Related/YourOutfit.jsx';
-import ProductCard from '../client/src/components/Related/ProductCard.jsx';
 const sampleProduct = require('./mocks.js').sampleProduct;
-const reviews = require('./mocks.js').reviews;
+const styles = require('./mocks.js').styles;
+const sampleReview = require('./mocks.js').sampleReview;
 const relatedProducts = require('./mocks.js').relatedProducts;
+const generateStars = require('./mocks.js').generateStars;
 
 beforeAll(() => {
-  nock('http://127.0.0.1:3000')
+  nock('http://localhost:3000')
     .defaultReplyHeaders({
       'access-control-allow-origin': '*',
     })
-    .get('/reviews/meta')
-    .reply(200, reviews);
+    .persist()
+    .get(/\/reviews\/meta\/.*/)
+    .query(true)
+    .reply(200, sampleReview);
+  nock('http://localhost:3000')
+    .defaultReplyHeaders({
+      'access-control-allow-origin': '*',
+    })
+    .persist()
+    .get(/\/products\/.*\/related/)
+    .reply(200, [11111, 11112, 11113, 11114, 71702]);
+  nock('http://localhost:3000')
+    .defaultReplyHeaders({
+      'access-control-allow-origin': '*',
+    })
+    .persist()
+    .get(/\/products\/.*\/styles/)
+    .reply(200, styles);
+  nock('http://localhost:3000')
+    .defaultReplyHeaders({
+      'access-control-allow-origin': '*',
+    })
+    .persist()
+    .get(/\/products\/.*/)
+    .reply(200, sampleProduct);
 });
 
-describe('ProductCard component', () => {
-  const renderComponent = () => {
-    return render(<ProductCard
-      product={sampleProduct}
-      generateStars={ function() { return 'stars'; }}
-      setIsFetching={ function() { return; }}
+const RelatedProductsComponent = () => {
+  return render(
+    <RelatedProducts
+      addToOutfit={() => { return; }}
+      yourOutfit={() => { return; }}
+      relatedProducts={relatedProducts}
+      generateStars={ generateStars }
+      isFetching={false}
+      setIsFetching={() => { return; }}
+      currentProduct={ relatedProducts[0] }
     />, {wrapper: Router});
```

Comment:
These tests look great! A couple of notes I have: 
1. I think they want you to break different component tests into different files, rather than having all of your tests in one file. That's obviously more of a nitpick thing.
2. I think you'll want to add some tests to make sure your widget doesn't break the app if no data is fed each component. This will also help you implement error handling for each of your components, so if for some reason API data doesn't come through, your widget won't kill the app. 
3. I remember Leslie saying something about not needing to test styling but rather functionality. In the future, I think you're safe to skip testing styles if it's not adding to functionality.
4. Another thing that you don't have to change, but might be helpful in the future writing tests: a different way to write this: `fireEvent(
      await screen.findByRole('button', {name: 'scroll right'}),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );`
that might not be so cumbersome is:
`const btn = await screen.findByRole('button', {name: 'scroll right'})
fireEvent.click(btn)` which does the same thing.

Reply:
Those are great suggestions, thank you! For now I think I'll save those for when I start to refactor my tests, just so I can get my (nearly) finished tests up on the test branch along with the new API fix.